src/keyedit.cpp

```diff
@@ -83,41 +84,75 @@ class KeyMapForm : public IntFormAnimated
 	}
 
 	void checkPushedKeyCombo();
-	bool pushedKeyCombo(KEY_CODE subkey);
+	bool pushedKeyCombo(const KeyMappingInput input);
 
 private:
 	std::shared_ptr<ScrollableListWidget> keyMapList;
+	std::shared_ptr<W_BUTTON> createKeyMapButton(const unsigned int id, const KeyMappingSlot slot, struct DisplayKeyMapData* targetFunctionData);
 	void unhighlightSelected();
 	void addButton(int buttonId, int y, const char *text);
 };
 
-struct DisplayKeyMapCache {
+struct DisplayKeyMapData {
+	explicit DisplayKeyMapData(void (* const function)(), const std::string name)
+		: mappings(std::vector<KEY_MAPPING*>(static_cast<unsigned int>(KeyMappingSlot::LAST), nullptr))
+		, function(function)
+		, name(name)
+	{
+	}
+
+	std::vector<KEY_MAPPING*> mappings;
+	void (*function)();
```

Comment:
The `std::function` refactor cannot properly be done here yet due to e.g. `keyGetMappingFromFunction` in `keymap.h/cpp` relying on raw function ptrs as mappable action identifiers. I think I've done this in #1508 already by using the save ID strings as ids instead of the functions, so this should still work fine for this PR?

Reply:
I'm supportive of leaving this for #1508. 👍